src/piadm.sh

```diff
@@ -512,10 +512,19 @@ install() {
 		# Explicit boot stamp or URL.
 
 		# Check if URL exists first
+		# XXX I think max-time 30s to fetch headers
+		# is more than enough.
 		vecho "Checking if URL $1 exists"
 		if ! "${CURL[@]}" --max-time 30 --connect-timeout \
 				10 --head "$1" > /dev/null 2>&1; then
-			vecho "URL $1 is not accessible or does not exist"
+			# Get HTTP status code for error reporting
+			http_code=$(curl -ks -w "%{http_code}" --max-time 30 --connect-timeout \
```

Comment:
This feels like one-too-many curl invocations...  when I big-picture review this again later today I'll figure out more.

Reply:
I'm imagining one less curl invocation (and my shell-fu may be weak and missing something...):

```
    if ! "${CURL[@]" --max-time 30 --connect-timeout \
        10 --head "$1" -w "%{http_code}" 2>/dev/null | tail -1 > /var/run/code.$$; then
        http_code=$(cat /var/run/code.$$)
        /bin/rm -f /var/run/code.$$
        ...
```